From: Russell King <rmk+kernel@armlinux.org.uk>
Date: Tue, 12 Jul 2016 16:45:43 +0100
Subject: [PATCH] phy: marvell: 88E1540: add flow control support

The Marvell PHYs support pause frame advertisments, so we should not be
masking their support off.  Add the necessary flag to the Marvell PHY
to allow any MAC level pause frame support to be advertised.

Signed-off-by: Russell King <rmk+kernel@armlinux.org.uk>
---

--- a/drivers/net/phy/marvell.c
+++ b/drivers/net/phy/marvell.c
@@ -1698,7 +1698,7 @@ static struct phy_driver marvell_drivers
 		.phy_id = MARVELL_PHY_ID_88E1540,
 		.phy_id_mask = MARVELL_PHY_ID_MASK,
 		.name = "Marvell 88E1540",
-		.features = PHY_GBIT_FEATURES,
+		.features = PHY_GBIT_FEATURES | SUPPORTED_Pause,
 		.flags = PHY_HAS_INTERRUPT,
 		.probe = marvell_probe,
 		.config_init = &marvell_config_init,
